src/opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+  Copyright 2019 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/parser/eclipse/EclipseState/Schedule/Group/GConSump.hpp>
+
+namespace Opm {
+
+GConSump::GConSump() {
+}
+
+
+bool GConSump::has(const std::string& name) const {
+    return (groups.find(name) != groups.end());
+}
+
+
+const GConSump::GCONSUMPGroup& GConSump::get(const std::string& name) const {
+
+    auto it = groups.find(name);
+    if (it == groups.end())
+        throw std::invalid_argument("Current GConSump obj. does not contain '" + name + "'.");
+    else
+        return it->second;
+}
+
+
+void GConSump::add(const std::string& name, const UDAValue& consumption_rate, const UDAValue& import_rate, const std::string network_node) {
+    groups[name] = GCONSUMPGroup();
+    GConSump::GCONSUMPGroup& group = groups[name];
```

Comment:
Is `add` supposed to clear out any existing values for `name`?  If not, then the above is wasteful.  The `groups` member is already a `std::map` so the index operator will default construct a `GCONSUMPGroup` object in `name`'s position if one does not already exist.

If you do not need/intend to clear existing values, then these two lines can be simplified to
```C++
GConSump::GConSump& group = this->groups[name];
```